Rewrited point update controller

De1taVee opened the pull request:
Added new functions in PointRestController class, edited point update function. Fixed missed deletedAt field in GroupDTO class, added setters for that field.

signifatee committed:
Rewrited point update controller

ptz2018 commented on src/main/java/com/example/projects/dto/GroupDTO.java:
```diff
@@ -50,4 +53,12 @@ public Date getUpdatedAt() {
     public void setUpdatedAt(Date updatedAt) {
         this.updatedAt = updatedAt;
     }
+
+    public Date getDeletedAt() {
+        return deletedAt;
+    }
+
+    public void setDeletedAt(Date deletedAt) {
+        this.deletedAt = deletedAt;
+    }
```
please revert

ptz2018 commented on src/main/java/com/example/projects/service/PointService.java:
```diff
@@ -4,5 +4,5 @@
 
 public interface PointService {
-    public PointDTO update(PointDTO point, int id);
+    public PointDTO update(PointDTO pointdto, Point point);
```
pointDTO

ptz2018 commented on src/main/java/com/example/projects/editors/PointidEditor.java:
```diff
@@ -0,0 +1,33 @@
+    public PointidEditor(GroupRepository groupRepository){
+        this.groupRepository = groupRepository;
+    }
+
+
```
please remove extra lines

ptz2018 commented on src/main/java/com/example/projects/editors/PointidEditor.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public void setAsText(String text) throws NumberFormatException{
+        Group group = groupRepository.getById(Integer.parseInt(text));
```
please throw custom exception

ptz2018 commented on src/main/java/com/example/projects/controllers/rest/PointRestController.java:
```diff
@@ -44,8 +56,8 @@ private GroupDTO getGroupById(@PathVariable int id){
-    private PointDTO updatePoint(@RequestBody PointDTO pointDTO, @PathVariable int id){
-        return pointService.update(pointDTO, id);
+    @PatchMapping("/{point}")
+    private PointDTO updatePoint(@PathVariable Point point, @RequestBody PointDTO pointDTO) {
```
public

signifatee committed:
Code cleanup;
Reverted deletedAt field, setters in GroupDTO;
Added custom exception for setAsText in PointidEditor class;
updatePoint() in PointRestController now is public
